### ISSUE TITLE
Copying state to S3 fails because init user can't assume TerragruntAdmin role

### ISSUE THREAD

**Issue by tPl0ch** [liatrio/aws-accounts-terraform#11]
All the organisation accounts are created and there is a assume role policy attached to the `terraform-init` user in the master account. The `init.sh` script fails though with the following error:

```
=== COPYING LOCAL STATE TO S3 ===
ERRO[0001] Error finding AWS credentials (did you set the AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY environment variables?): AccessDenied: User: arn:aws:iam::239042211XXX:user/terraform-init is not authorized to perform: sts:AssumeRole on resource: arn:aws:iam::"289486821XXX":role/TerragruntAdministrator
        status code: 403, request id: 294b7bd3-bf03-4f78-bb00-823133510185 
ERRO[0001] Unable to determine underlying exit code, so Terragrunt will exit with error code 1 
```

The AWS environment variables and default profile are set. I checked the policies and there is a trust policy attached to `terraform-init`:

```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "AssumeInfosecTerragruntAdministratorRole",
            "Effect": "Allow",
            "Action": "sts:AssumeRole",
            "Resource": "arn:aws:iam::289486821XXX:role/TerragruntAdministrator"
        }
    ]
}
```

So right now the only thing I could imagine is a missing trust relationship on the infosec account side to allow the `terraform-init` to assume roles there. But I might be wrong.

Thanks for any advice on this.

**Comment by tonycpsu** [comment]
Same issue here.  @tPl0ch did you find a solution?

**Comment by tonycpsu** [comment]
@jphayes any insight as to what might be going on here?  I'd love to make use of this but haven't been able to figure out why it won't allow the role assumption even though both roles seem to have the right policies attached.

**Comment by tPl0ch** [comment]
@tonycpsu Yes, if you look closely at the resource string `arn:aws:iam::"289486821XXX":role/TerragruntAdministrator` you see that the quotes are added to the raw string. This is a behavioural change in terraform 0.14, see https://github.com/hashicorp/terraform/issues/26831

The fix for me was changing the lines containing `$(terraform output output_key)` in `init.sh` to `$(terraform output -json | jq -r .output_key.value)`. This should probably go into this repository since the terraform devs are pointing towards using the JSON representation for automation.

**Comment by tonycpsu** [comment]
@tPl0ch: nice, thanks for the tip.  Was able to get it working.  If the maintainers think it's worthwhile, I'll be happy to submit a PR.

**Comment by TizianoPerrucci** [comment]
Instead of parsing the json output you can also get the raw value `$(terraform output -raw output_key)`

but indeed since terraform 0.14 the `init.sh` needs a fix
